pkg/tsdb/elasticsearch/data_query.go

```diff
@@ -41,12 +42,15 @@ var newElasticsearchDataQuery = func(ctx context.Context, client es.Client, data
 
 func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
 	start := time.Now()
+	response := backend.NewQueryDataResponse()
 	e.logger.Debug("Parsing queries", "queriesLength", len(e.dataQueries))
 	queries, err := parseQuery(e.dataQueries, e.logger)
 	if err != nil {
 		mq, _ := json.Marshal(e.dataQueries)
 		e.logger.Error("Failed to parse queries", "error", err, "queries", string(mq), "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
-		return &backend.QueryDataResponse{}, err
+		err := errorsource.PluginError(err, false)
+		response.Responses[e.dataQueries[0].RefID] = errorsource.Response(err)
+		return response, err
```

Comment:
looks like we are repeating ourselves here.  maybe just create an `assignSource` function

```
func assignSource(refID string, response backend.QueryDataResponse, err error) (backend.QueryDataResponse, error) {
  err := errorsource.PluginError(err, false)
  response.Responses[refID] = errorsource.Response(err)
  return response, err
```

then just 
```
return assignSource(e.dataQueries[0].RefID, response, err)
```